google/datalab/contrib/mltoolbox/commands/_ml.py

```diff
@@ -0,0 +1,110 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Google Cloud Platform library - ml cell magic."""
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+try:
+  import IPython
+  import IPython.core.display
+  import IPython.core.magic
+except ImportError:
+  raise Exception('This module can only be loaded in ipython.')
+
+import pandas as pd
+import numpy as np
+import six
+
+import google.datalab.utils.commands
+import google.datalab.contrib.mltoolbox._local_predict as _local_predict
+
+
+@IPython.core.magic.register_line_cell_magic
+def ml(line, cell=None):
+  """Implements the datalab cell magic for MLToolbox operations.
+
+  Args:
+    line: the contents of the ml command line.
+  Returns:
+    The results of executing the cell.
+  """
+  parser = google.datalab.utils.commands.CommandParser(
+      prog='%ml',
+      description="""
+Execute MLToolbox operations.
+
+Use "%ml <command> -h" for help on a specific command.
+""")
+
+  predict_parser = parser.subcommand(
+      'predict', help='Predict with local or deployed models. ' +
+      'Prediction data is provided in cell in yaml format of CSV lines. For example, ' +
+      """
+prediction_data:
+  - value1,value2, ...
+  - value1,value2, ...
+""")
+  predict_parser.add_argument('--model', required=True,
+                              help='The model path if not --cloud, or the id in ' +
+                                   'the form of model.version if --cloud.')
+  predict_parser.add_argument('--headers', required=True,
+                              help='The comma seperated headers of the prediction data.')
+  predict_parser.add_argument('--image_columns',
+                              help='Comma seperated headers of image URL columns. ' +
+                                   'Required if prediction data contains image columns.')
+  predict_parser.add_argument('--show_image', action='store_true', default=True,
+                              help='Whether to add a column of images in output.')
+  predict_parser.add_argument('--data',
+                              help='The prediction data defined in other cells. ' +
+                                   'It can be a DataFrame, a list of CSV lines or a list ' +
+                                   'of dictionary. If not provided, then cell input is used ' +
+                                   'as a list of CSV lines')
+
+  predict_parser.set_defaults(func=_predict)
+
+  return google.datalab.utils.commands.handle_magic_line(line, cell, parser)
+
+
+def _predict(args, cell):
+  headers = args['headers'].split(',')
+  img_cols = args['image_columns'].split(',') if args['image_columns'] else []
+  env = google.datalab.utils.commands.notebook_environment()
+  cell_data = google.datalab.utils.commands.parse_config(cell, env)
+  if 'prediction_data' not in cell:
```

Comment:
I think you need to be checking --data first, and if it is missing then inspect the cell